internal/app/api/v1/executions.go

```diff
@@ -569,6 +571,18 @@ func (s *TestkubeAPI) streamLogsFromJob(executionID string, w *bufio.Writer) {
 	}
 }
 
+func (s TestkubeAPI) getNextExecutionNumber(testName string) int {
+	execution, err := s.ExecutionResults.GetLatestByTest(context.Background(), testName, "number")
```

Comment:
it will work only if we don't have parallel runnng tests

Reply:
It should be done on top of some atomic counter, e.g. by using Mongos' findAndModify method on top of some counters collection, similar like posgresql sequences work.

Reply:
Done follow up task ✅